lib/neovim-plugin.nix

```diff
@@ -161,8 +162,43 @@
                   # Add the plugin setup code `require('foo').setup(...)` to the lua configuration
                   (lib.optionalAttrs callSetup { ${namespace}.${name}.luaConfig.content = setupCode; })
 
-                  # Write the lua configuration `luaConfig.content` to the config file
-                  (setLuaConfig cfg.luaConfig.content)
+                  # Write the lua configuration `luaConfig.content` to the config file when lazy loading is not enabled
+                  (lib.mkIf (!cfg.lazyLoad.enable) (setLuaConfig cfg.luaConfig.content))
+
+                  # When lazy loading is enabled for this plugin, route its configuration to the enabled provider
+                  (lib.mkIf cfg.lazyLoad.enable {
+                    assertions = [
+                      {
+                        assertion = (isColorscheme && colorscheme != null) || cfg.lazyLoad.settings != { };
+                        message = "You have enabled lazy loading for ${originalName} but have not provided any configuration.";
+                      }
+                      {
+                        assertion = cfg.lazyLoad.enable && (config.plugins.lz-n.enable || config.plugins.lazy.enable);
+                        message = "You have enabled lazy loading for ${originalName} but have not enabled any lazy loading plugins.";
+                      }
+                    ];
+                    plugins.lz-n = lib.mkIf config.plugins.lz-n.enable {
+                      plugins = [
+                        (
+                          {
+                            __unkeyed-1 = originalName;
+                            # Use provided after, otherwise fallback to normal function wrapped lua content
+                            after =
+                              let
+                                after = cfg.lazyload.settings.after or null;
+                                default = ("function()\n " + cfg.luaConfig.content + " \nend");
+                              in
+                              if lib.isString after then after else default;
```

Comment:
shouldn't we check if after is defined? If a user does `after = lib.mkRaw { .... }`  this will be surprising

Reply:
Good point re-raw types... However I don't think simply checking it is defined is enough...

My intention was to check if defined via `or null`, and exclude unexpected types (permitted by `freeform = attrsOf anything`) using `isString`.

I'm also unsure whether blindly passing a raw type through to `lz-n.plugins.*.after` may or may not be handled correctly?

If we can handle raw, then maybe we want to check "`isString` or `types.rawLua.check`"? Or if we're being strict about not coercing strings->lua, maybe we should only accept rawLua?